daphne/src/messages/mod.rs

```diff
@@ -522,20 +521,6 @@ impl Default for BatchSelector {
     }
 }
 
-impl TryFrom<Query> for BatchSelector {
-    type Error = DapError;
-
-    fn try_from(query: Query) -> Result<Self, DapError> {
-        match query {
-            Query::TimeInterval { batch_interval } => Ok(Self::TimeInterval { batch_interval }),
-            Query::FixedSizeByBatchId { batch_id } => Ok(Self::FixedSizeByBatchId { batch_id }),
-            Query::FixedSizeCurrentBatch => Err(fatal_error!(
-                err = "tried to make a BatchSelector from a FixedSizeCurrentBatch query",
-            )),
-        }
-    }
-}
```

Comment:
Reviewer note: We expect to hit this error even when things are fine, but `fatal_error` logs the error. I've replaced with `Query::into_batch_sel()`, defined below.